### ISSUE TITLE
ansible task executor should print the result information when json.load rasie an ValueError

### ISSUE THREAD

**Issue by Jack47** [ansible/ansible#16133]
##### ISSUE TYPE

- Feature Idea

Could ansible task executor print the result information when `json.load` rasie an ValueError?
It will save programer's time and make life happier.

I spent a whole day and finally solved the problem I met as illustrated below. I could solved it in minutes if ansible task executor has printed the result information when `json.load` rasie an ValueError.
##### ANSIBLE VERSION

```
ansible --version
ansible 2.0.2.0
  config file =
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

MacOS 10.9
##### SUMMARY

I want to launch shell command to build docker image asynchronously, the build is success but this error occured:
`fatal: [192.168.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "The async task did not return valid JSON: No JSON object could be decoded"}`

Although it explained that the async task return a invalid json, but **_I don't know the returned result**_.

I modified the `executor/task_executor.py` to print the result, and I found there is something wrong with my term's locale setting

`['stdout': u'/etc/profile.d/lang.sh: line 19: warning: setlocale: LC_CTYPE: cannot change locale (UTF-8): No such file or directory\\r\\n{\"started\": 1, \"results_file\": \"/root/.ansible_async/177959462548.25553\", \"ansible_job_id\": \"177959462548.25553\"}\\r\\n'](url)`

I solved the locale problem with the help of this [link](http://www.cyberciti.biz/faq/os-x-terminal-bash-warning-setlocale-lc_ctype-cannot-change-locale/), and this problem didn't occur.

**Comment by alikins** [comment]
I think this should probably be split into two issues:

1) bug report/behavior change request for showing more useful info on the failures. 
    The needs fixing/improving, but it is not unique to (2)

2) bug report referencing http://www.cyberciti.biz/faq/os-x-terminal-bash-warning-setlocale-lc_ctype-cannot-change-locale/ 

  Not sure yet how ansible could workaround this, but it's obtuse enough to be it's own issue. If nothing else it's a good candidate for info to include in the documentation.

I'm going to leave this issue open and consider it (1) above. I'll open a new issue for (2) and reference here.

**Comment by mattclay** [comment]
PR #16534 been merged and may fix your issue. The problem may go away entirely with this fix. If not, the error message "The async task did not return valid JSON" you reported should be replaced with a more informative error, allowing further troubleshooting.

Unless further tests reveal major bugs, this fix will be included in the upcoming 2.1.1 release. If you'd like to test and confirm the fix before then, please follow the instructions on [running from source](http://docs.ansible.com/ansible/intro_installation.html#running-from-source). Be sure to use `-b stable-2.1` when you clone so you are testing the stable-2.1 branch.

If we don't hear back before the 2.1.1 release, we'll consider this issue resolved and close the issue. If the problem is not resolved, please update this issue with the relevant details.

**Comment by mattclay** [comment]
Closing as resolved with the 2.1.1 release.
